### PR TITLE
Export auth server version for upgrader

### PR DESCRIPTION
**Pull request by bernardjkim**
Paired with https://github.com/gravitational/teleport.e/pull/2722
Supports https://github.com/gravitational/cloud/issues/6773

Teleport will now export the auth server version:
- On kube agents, the auth server version will be exported to the backend with key `agent-auth-version`.
- On systemd agents, the auth server version will be exported to the `/etc/teleport-upgrade.d/version` unit file.

This is necessary for the teleport-upgrade script to be able to identify the auth server version, and prevent an upgrade if the target upgrade version is a newer major version than the auth server.

### CODE REVIEW

File: lib/versioncontrol/upgradewindow/upgradewindow.go [hunk 1]
```diff
@@ -40,16 +40,25 @@ const (
...
 	kubeSchedKey = "agent-maintenance-schedule"
 
+	// kubeVersionKey is the key under which the kube controller version is exported
+	kubeVersionKey = "agent-auth-version"
+
 	// unitScheduleFile is the name of the file to which the unit schedule is exported.
 	unitScheduleFile = "schedule"
 
+	// unitVersionFile is the name of the file to which the version is exported.
+	unitVersionFile = "version"
```
Comment (espadolini) [inline lib/versioncontrol/upgradewindow/upgradewindow.go:50]: `version` feels way too generic, maybe `auth-version`?

```suggestion
	unitVersionFile = "auth-version"
```

**Review by espadolini — COMMENTED** [review]
The auth version is never going to change while we are holding a connection to the auth server, and we're holding a connection to the auth server as part of the standard behavior of the inventory control stream; could we just update the auth version whenever that connects, instead of piggybacking onto the upgrade window synchronizer? `DownstreamInventoryHello` already has the version of the auth server in it.

File: lib/versioncontrol/upgradewindow/upgradewindow.go [hunk 2]
```diff
@@ -427,6 +477,18 @@ func (e *systemdDriver) Sync(ctx context.Context, rsp proto.ExportUpgradeWindows
 	return nil
 }
 
+func (e *systemdDriver) SyncAuthVersion(ctx context.Context, version string) error {
+	if err := os.MkdirAll(e.cfg.ConfigDir, defaults.DirectoryPermissions); err != nil {
+		return trace.Wrap(err)
+	}
+
+	if err := os.WriteFile(e.versionFile(), []byte(version), defaults.FilePermissions); err != nil {
```
Comment (espadolini) [inline lib/versioncontrol/upgradewindow/upgradewindow.go:485]: @fspmarshall shouldn't this (and the schedule one) be atomic writes? What happens if the upgrader runs right as we're writing the file?

**Comment by bernardjkim** [comment]
Closing in favor of https://github.com/gravitational/teleport/pull/35150

**Pull request closed without merging**